### PR TITLE
HSI Tool

### PR DESCRIPTION
**Pull request by annarieger**
This PR introduces HSI tool which enables showing of feature properties via hovering on configured map layers.

image

Please review @terrestris/devs

### CODE REVIEW

File: packages/baseclient-components/src/component/Menu/FeatureInfo/FeatureInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+  /**
+   * Dispatch function
+   */
+  dispatch: (arg: any) => void
+}
+
+interface FeatureInfoState {
+  menuHidden: boolean,
+  gridWinHidden: boolean,
+  featuresToShow: any[], //OlFeature[]
```
Comment (hwbllmnn) [inline packages/baseclient-components/src/component/Menu/FeatureInfo/FeatureInfo.tsx]:
```suggestion
  featuresToShow: any[],
```

File: packages/baseclient-components/src/component/Popup/FeatureInfo/FeatureInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+      while (this.overlayHelperElement.firstChild) {
+        this.overlayHelperElement.removeChild(this.overlayHelperElement.firstChild);
+      }
+      // Create a clone of the feature-info-popup element to prevent any side
+      // effects with react or the ol overlay.
+      const overlayElementClone: any = this.overlayElement.cloneNode(true);
+      // Append the cloned element to the helper elemend, this way it will be
+      // mounted and size attributes are available.
+      const mount: any = this.overlayHelperElement.appendChild(overlayElementClone);
+      // We assume, taht as soon as the element has a `clientHeight` set, it
```
Comment (hwbllmnn) [inline packages/baseclient-components/src/component/Popup/FeatureInfo/FeatureInfo.tsx]:
```suggestion
      // We assume that as soon as the element has a `clientHeight` set, it
```

File: packages/baseclient-state/src/actions/RemoteFeatureAction.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+ * @param {Object} fetchOpts The options to apply to the fetch,
+ *                           see https://github.github.io/fetch/.
+ * @param {ol.format} format The ol.format instance to read the fetched features
+ *                           with.
+ * @param {Object} readerOpts The options to apply to the readFeatures() method
+ *                            provided by the ol.format instance.
+ * @return {Function} The thunk.
+ */
+export function fetchFeaturesFromResource(url: string, fetchOpts: any, format: any, readerOpts: any) {
+  return (/*dispatch*/) => {
```
Comment (hwbllmnn) [inline packages/baseclient-state/src/actions/RemoteFeatureAction.ts]:
```suggestion
  return () => {
```

File: packages/baseclient-components/src/component/Menu/FeatureInfo/FeatureInfo.tsx (hunk 1 above)
Comment (annarieger) [reply packages/baseclient-components/src/component/Menu/FeatureInfo/FeatureInfo.tsx]: I'd like to leave this comment here as reminder to change the type after typescript based `ol6` version is released and we finally can use "real" types for openlayers objects.

There are much more similar comments across the code btw.

File: packages/baseclient-components/tsconfig.build.json [hunk 4]
```diff
@@ -14,10 +14,10 @@
     "moduleResolution": "node",
     "rootDir": "src",
     "forceConsistentCasingInFileNames": true,
-    "noImplicitReturns": true,
+    "noImplicitReturns": false,
     "noImplicitThis": true,
     "noImplicitAny": true,
-    "strictNullChecks": true,
+    "strictNullChecks": false,
```
Comment (weskamm) [inline packages/baseclient-components/tsconfig.build.json]: :-1:

**Comment by weskamm** [comment]
Thanks for getting this in!
Just a little sad we had to disable `strictNullChecks`..

**Pull request merged**